### DISCUSSION TITLE
Connection to eos on nomad on same PC

### DISCUSSION THREAD

**Discussion by banditito** [sstaub/eosstreamdeckplus#3] [Q&A]
Hello, 
I' tried to connect the elgato stream deck plus to my eos nomad on laptop windows 10.
i've upload your file on the stream deck via companion. 
I'm not yet connect to artnet.
so my nomad is just in Wifi . 
How to connect the stream deck to nomad ?? 
Thanks for your answer, i begin with stream deck companion and osc.

**Comment by sstaub** [comment]
May be this a problem of using DHCP. Your laptop connection should use a fixed IP address.

**Reply by banditito** [reply]
Ok .
my laptop isn't connected to a artnet network at home ... is it a problem ?

**Comment by sstaub** [comment]
It has nothing to do with ArtNet. You must configure your OSC setting on Nomad and also Companion. To work Companion on the same laptop as Nomad you need an active network connection (LAN or WLAN).

**Reply by banditito** [reply]
yes i understand this. The connection must have internet acces ?
Because my lan (home) is in WIfi . I have configure ipv4 in 192.168.1.30 an d mask 255.255.255.0. DNS pref 1.1.1.1 aux 1.0.0.1.
Sorry i'm not expert in lan and wlan with osc ..

I have in bitfocus companion, 2 osc connection : 1 osc generic and the other etc-osc.
1 : 192.168.1.30 port 8001
2: 192.168.1.30 user id 1

where is my mistake lol ?

**Comment by sstaub** [comment]
No internet connection needed. Are the Nomad settings OK (UDP RX Port should 8001)?
If the connection is stable the connections in companion should green.
TCP setting must OSC v1.0 for etc-eos

**Reply by banditito** [reply]
I don't understand tcp setting must osc v1.0 for etc-eos??
but it's ok now, i've forget the udp rx port 8001 in nomad.

**Comment by sstaub** [comment]
The TCP Protocol type must set to 1.0 in the Shell network tab.

**Reply by banditito** [reply]
Ok it's ok it work fine now... 
2 other questions about pan and tilt .. We don't want the fine parameter.. just pan and tilt ..whats the osc path and value to modify ?
And about pages : I want to add one more page with some master .. how to do this? 
Many thanks a lot for your help ..

**Comment by sstaub** [comment]
You can simple delete the label buttons / encoder. The extended version use the label button to switch between fine/tilt and acceleration..
To add pages it is a little bit tricky, maybe you can reconfigure an existing page which you don't need.

**Reply by banditito** [reply]
Ok ..
maybe if the 2.19  button is a link to a master page (page 10 for example??) and on the page 10 the last buttton a link to page 1 ??
what would be the syntax, if it is possible ??

**Comment by sstaub** [comment]
There is a 'Set surface with index to page' action, like the other buttons. Have also a look to the documentation page of companion.

**Reply by banditito** [reply]
Ok thanks a lot i'll read this before ...

**Comment by eklighting** [comment]
Good afternoon, I have tried to use your companion file and everything works great except I don't get anything on the encoders. So when I select focus, I don't get pan and tilt to come up to be able to adjust the fixture. Is there something I need to adjust on the code?

Thank you for your help

**Comment by sstaub** [comment]
Which EOS / Companion version? Is the connection status ok?

**Reply by eklighting** [reply]
Connection status is all good, latest version of both companion 3.2.0 and EOS Nomad 3.1.5.8. as I say all other commands work, but when  you go to the any of the encoders you dont get the pan and tilt for instance, they are just blank. firing cues etc is all good. Thank you for you help and for such a fantatsic bit of coding

**Reply by sstaub** [reply]
Is this only with pan / tilt. Maybe you can try the original example https://github.com/bitfocus/companion-module-etc-eos
Did you tried with EOS 3.2.5?

**Reply by eklighting** [reply]
I will update EOS and see if that makes a difference. No this isnt just with Pana dn Tilt it is with all the encoders, so color, form, etc.
Thank you for your help
